Automattic/jetpack
# Improve Story block media loading

Addresses part of https://github.com/Automattic/jetpack/issues/16591:
- Show a progress indicator when uploading media from the block editor, prevent story playback while its uploading
- Improve loading of media files
- Better fit media on the player, use heuristics to crop some part of the image to make it fit a user screen (for instance allow 10% cropping max, after which margins can be displayed)
- Refine UX
- Scale down story embeds (make it 50% smaller?)
 
With this change we stop rendering all the slides dynamically from the server. Instead we only render the first one and serialize the media into the markup so the client can retrieve it and load slides dynamically. Slides are rendered as the story is played, with a preload of the next slide.

It also applies the latest changes in design regarding stories:
- Smaller embeds
- Shadow and animation on hover
- Play and replay button removed, replaced with transparent overlay
- Displaying an expand icon on the top right

This doesn't update buttons to use `@wordpress/element` nor fixes accessibility issues, those are tackled in this PR https://github.com/Automattic/jetpack/pull/16852

Here is how it looks like:

Frontend:
<img width="639" alt="Screenshot 2020-08-10 at 15 27 37" src="https://user-images.githubusercontent.com/230230/89787621-13729680-db1e-11ea-9e4e-fe33052c3310.png">

Block Editor:
<img width="591" alt="Screenshot 2020-08-10 at 15 27 08" src="https://user-images.githubusercontent.com/230230/89787652-1ff6ef00-db1e-11ea-96e9-c61567e192f0.png">

#### Changes proposed in this Pull Request:
In this PR we lazy load media while stories are being played. This way, we improve the initial page load speed.
We also define a list of alternative images for each slide using srcset and force the browser to load the right image for the current viewport.

#### Does this pull request change what data or activity we track or use?
No

#### Testing instructions:
Load a page with a story in it. Notice in the network tab only the fist slide is loaded.
Disable JS and reload the page, notice the story looks almost identical as the version with JS

#### Proposed changelog entry for your changes:
-  Improve loading of media files for the Story block
